packages/lesswrong/server/fmCrosspost/resolvers.ts

```diff
@@ -120,14 +128,19 @@ const crosspostResolvers = {
     },
   },
   Query: {
-    getCrosspost: async (_root: void, { args }: { args: GetCrosspostRequest }) => {
-      const crosspostData = await makeCrossSiteRequest(
-        'getCrosspost',
-        args,
-        'Failed to get crosspost'
-      );
-
-      return crosspostData.document;
+    getCrosspost: async (_root: void, {args}: {args: GetCrosspostRequest}) => {
+      const key = stringify(args);
+      let promise = foreignPostCache.get(key);
+      if (!promise) {
+        promise = makeCrossSiteRequest(
+          'getCrosspost',
+          args,
+          'Failed to get crosspost'
+        );
+        foreignPostCache.set(key, promise);
```

Comment:
This took me a minute to understand, but it's really cool that you're setting the _promise_ as the value of the cache.